simulators/portal-interop/src/main.rs

```diff
@@ -480,3 +492,59 @@ dyn_async! {
         }
     }
 }
+
+dyn_async! {
+    // test that a node will return a receipts via FINDCONTENT over uTP that it has stored locally
+    async fn test_find_content_receipts_over_utp<'a> (client_a: Client, client_b: Client) {
+        let receipts_key: HistoryContentKey = serde_json::from_value(json!(RECEIPTS_KEY)).unwrap();
+        let receipts_value: HistoryContentValue = serde_json::from_value(json!(RECEIPTS_VALUE)).unwrap();
+
+        let header_with_proof_key: HistoryContentKey = serde_json::from_value(json!(HEADER_WITH_PROOF_KEY)).unwrap();
+        let header_with_proof_value: HistoryContentValue = serde_json::from_value(json!(HEADER_WITH_PROOF_VALUE)).unwrap();
+
+        match client_b.rpc.store(receipts_key.clone(), receipts_value.clone()).await {
+            Ok(result) => if !result {
+                panic!("Error storing header with proof for Receipts verification");
+            },
+            Err(err) => {
+                panic!("Error storing header with proof for Receipts verification: {err:?}");
+            }
+        }
+
+        match client_b.rpc.store(header_with_proof_key.clone(), header_with_proof_value.clone()).await {
+            Ok(result) => if !result {
+                panic!("Unable to store header with proof for find content immediate return test");
+            },
+            Err(err) => {
+                panic!("Error storing header with proof for find content immediate return test: {err:?}");
+            }
+        }
+
+        let target_enr = match client_b.rpc.node_info().await {
+            Ok(node_info) => node_info.enr,
+            Err(err) => {
+                panic!("Error getting node info: {err:?}");
+            }
+        };
+
+        match client_a.rpc.find_content(target_enr, receipts_key.clone()).await {
+            Ok(result) => {
+                match result {
+                    // todo: when clients conform with spec we will need to add a check if
```

Comment:
I think it is fine to add a test that clients fail on, if it is according to spec

Reply:
It is a field in the json-rpc find_content spec I believe

Reply:
Yeah, that's what I mean. It's already in the spec, so you can activate a test for it. If clients don't support it, it creates somewhat of an incentive to fix it :)